nnf/__init__.py

```diff
@@ -998,6 +998,53 @@ def pair(node: NNF) -> NNF:
 
         return pair(sentence)
 
+    def project(self: T_NNF, names: 't.FrozenSet[Name]') -> 'NNF':
+        """Dual of :meth:``forget``: will forget all variables not given"""
+        return self.forget(self.vars() - names)
+
+    def forget_aux(self: T_NNF) -> 'NNF':
+        """Returns a theory that forgets all of the auxillary variables"""
+        aux_vars = frozenset({v for v in self.vars() if isinstance(v, Aux)})
+        return self.forget(aux_vars)
+
+    def forget(self: T_NNF, names: 't.FrozenSet[Name]') -> 'NNF':
+        """Forget a set of variables from the theory.
+
+        Has the effect of returning a theory without the variables provided,
+        such that every model of the new theory has an extension (i.e., an
+        assignment) to the forgotten variables that is a model of the original
+        theory.
+
+        :param names: A frozenset of the variable names to be forgotten
+        """
+
+        if self.decomposable():
+            return self._forget_with_subs(names)
+        else:
+            return self._forget_with_shannon(names)
+
+    def _forget_with_subs(self: T_NNF, names: 't.FrozenSet[Name]') -> 'NNF':
```

Comment:
Hm, could this be implemented as `return self.condition({name: True for name in names}).simplify()`?

Reply:
Not quite, but it's worth having you look at it again: we need both positive and negative literals to be replaced with `true`. I'm assuming that's what the first check does.